framework/src/modules/interoperability/utils.ts

```diff
@@ -106,3 +106,16 @@ export const computeValidatorsHash = (
 
 export const sortValidatorsByBLSKey = (validators: ActiveValidators[]) =>
 	validators.sort((a, b) => a.blsKey.compare(b.blsKey));
+
+export const swapChainIDs = (ccm: CCMsg) => {
+	const newCcm = { ...ccm };
+
+	// Swap sendingChainID and receivingChainID of CCM without intermediary variable
+	/* eslint-disable no-bitwise */
+	newCcm.sendingChainID ^= newCcm.receivingChainID;
+	newCcm.receivingChainID ^= newCcm.sendingChainID;
+	newCcm.sendingChainID ^= newCcm.receivingChainID;
+	/* eslint-enable no-bitwise */
+
+	return newCcm;
+};
```

Comment:
you can also do below without using intermediary variable,
```ts
export const swapChainIDs = (ccm: CCMsg) => ({ ...ccm, receivingChainID: sendingChainID, sendingChainID: receivingChainID });
```

Reply:
I don't understand, how this code can be logical without an intermediary value under the hood of V8? If you set `receivingChainID` to something, then you would set `sendingChainID` to newly set `receivingChainID` if it is changed, unless you keep `receivingChainID`'s value in an intermediary value.

Reply:
Sorry to say intermediary here, I meant in the code we are writing not under the v8 hood. we can do it with spread operator (I think it simply overwrites the values given by us) instead of using bitwise here.

I used the same code and you can see the performance is better with spread and does the same job

Result:
"bitwise: 0.164ms"
"spread: 0.049ms"
```ts

const swapChainIDs = (ccm) => {
	const newCcm = { ...ccm };

	// Swap sendingChainID and receivingChainID of CCM without intermediary variable
	/* eslint-disable no-bitwise */
	newCcm.sendingChainID ^= newCcm.receivingChainID;
	newCcm.receivingChainID ^= newCcm.sendingChainID;
	newCcm.sendingChainID ^= newCcm.receivingChainID;
	/* eslint-enable no-bitwise */

	return newCcm;
};

const swapChainIDs2 = (ccm) => ({ ...ccm, receivingChainID: ccm.sendingChainID, sendingChainID: ccm.receivingChainID });

console.time('bitwise')
swapChainIDs({ sendingChainID: 1, receivingChainID: 2})
console.timeEnd('bitwise')

console.time('spread')
swapChainIDs2({ sendingChainID: 1, receivingChainID: 2})
console.timeEnd('spread')
```

Reply:
Thinking that also memory increase wouldn't be much, I'll apply your suggestion, it's more readable. Thanks!